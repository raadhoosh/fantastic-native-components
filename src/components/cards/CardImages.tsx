import * as React from "react";
import { StyledCardImages, StyledCardImagesTitle, StyledImage, StyledCardImagesText, StyledCardImagesFooter } from "./Cards.style";
import { ViewStyle, ImageSourcePropType } from "react-native";
import { Theme } from '..';

interface IProps {
  style?: ViewStyle | object | Array<ViewStyle>;
  primary?: boolean;
  secondary?: boolean;
  success?: boolean;
  info?: boolean;
  warning?: boolean;
  danger?: boolean;
  light?: boolean;
  dark?: boolean;
  inverse?: boolean;
  backgroundColor?: string;
  borderColor?: string;
  color?: string;
  width?: number | string;
  height?: number | string;
  imageWidth?: number | string;
  imageHeight?: number | string;
  theme?: Theme;
  onPress?: () => void;
  disabled?: boolean;
  fontSize?: string | number;
  borderRadius?: string;
  title?: string;
  text?: string;
  center?: boolean;
  titleColor?: string;
  source: ImageSourcePropType;
}

const CardImages = (props: IProps) => {

  return (
    <StyledCardImages {...props} activeOpacity={0.8} onPress={props.disabled ? undefined : props.onPress} >

      <StyledImage source={props.source} imageWidth={props.imageWidth} imageHeight={props.imageHeight} />
      <StyledCardImagesFooter {...props}>
        <StyledCardImagesTitle titleColor={props.titleColor} center={props.center} {...props}>
          {props.title}
        </StyledCardImagesTitle>
        <StyledCardImagesText color={props.color} center={props.center} {...props}>
          {props.text}
        </StyledCardImagesText>
      </StyledCardImagesFooter>
    </StyledCardImages>
  );
};

export default CardImages;
