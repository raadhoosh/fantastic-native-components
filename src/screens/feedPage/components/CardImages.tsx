import * as React from "react";
import { Row, CardImages } from "../../../components";
import { Dimensions } from "react-native";
const deviceWidth = Dimensions.get("window").width - 70;
type Item = {
    id: string;
    title: string;
    type: string;
    short_description: string;
    short_content ?: string;
    duration ?: string;
    url: string;
};
interface IProps {
    item: Item;
}
class App extends React.PureComponent<IProps> {
    render() {
        const { item } = this.props;
        return (
            <Row style={{ marginLeft: 12, marginRight: 12 }} >
                <CardImages
                    borderColor={"#000"}
                    source={{ uri: item.url }}
                    primary
                    titleColor={"#FF005A"}
                    title={item.title}
                    text={item.short_description}
                    width={deviceWidth} height={250} />
            </Row>
        );
    }
}

export default App;