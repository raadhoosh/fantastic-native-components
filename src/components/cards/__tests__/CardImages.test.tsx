import CardImages from "../CardImages";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
// tslint:disable-next-line:no-var-requires
const uri = require("../../../screens/imagePage/img_avatar3.png");
const createTestProps = (props: Object) => ({
  ...props,
});

describe("CardImages", () => {
  describe("rendering", () => {
    let component: ShallowWrapper;
    let props: Object;
    beforeEach(() => {
      props = createTestProps({
        color: "#f00",
        primary: true,
      });
      component = shallow(<CardImages source={uri} {...props}
        onPress={() => {
          alert("You tapped the CardImages!");
      }}
      >CardImages</CardImages>);
    });
    it("should render a <CardImages />", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
