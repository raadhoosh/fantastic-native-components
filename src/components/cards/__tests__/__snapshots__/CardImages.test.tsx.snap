// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImages rendering should render a <CardImages /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardImages
    color="#f00"
    onPress={[Function]}
    primary={true}
    source={
      Object {
        "testUri": "../../../src/screens/imagePage/img_avatar3.png",
      }
    }
  >
    CardImages
  </CardImages>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeOpacity": 0.8,
      "children": Array [
        <ForwardRef
          source={
            Object {
              "testUri": "../../../src/screens/imagePage/img_avatar3.png",
            }
          }
        />,
        <ForwardRef
          color="#f00"
          onPress={[Function]}
          primary={true}
          source={
            Object {
              "testUri": "../../../src/screens/imagePage/img_avatar3.png",
            }
          }
        >
          <ForwardRef
            color="#f00"
            onPress={[Function]}
            primary={true}
            source={
              Object {
                "testUri": "../../../src/screens/imagePage/img_avatar3.png",
              }
            }
          />
          <ForwardRef
            color="#f00"
            onPress={[Function]}
            primary={true}
            source={
              Object {
                "testUri": "../../../src/screens/imagePage/img_avatar3.png",
              }
            }
          />
        </ForwardRef>,
      ],
      "color": "#f00",
      "onPress": [Function],
      "primary": true,
      "source": Object {
        "testUri": "../../../src/screens/imagePage/img_avatar3.png",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "imageHeight": undefined,
          "imageWidth": undefined,
          "source": Object {
            "testUri": "../../../src/screens/imagePage/img_avatar3.png",
          },
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "rules": Array [
              "
    border-radius: ",
              [Function],
              ";
    width: ",
              [Function],
              ";
    height: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "Styled(Component)",
          "getSize": [Function],
          "prefetch": [Function],
          "propTypes": Object {
            "accessibilityLabel": [Function],
            "accessible": [Function],
            "blurRadius": [Function],
            "capInsets": [Function],
            "defaultSource": [Function],
            "onError": [Function],
            "onLayout": [Function],
            "onLoad": [Function],
            "onLoadEnd": [Function],
            "onLoadStart": [Function],
            "onPartialLoad": [Function],
            "onProgress": [Function],
            "resizeMethod": [Function],
            "resizeMode": [Function],
            "source": [Function],
            "style": [Function],
            "testID": [Function],
          },
          "render": [Function],
          "resolveAssetSource": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              color="#f00"
              onPress={[Function]}
              primary={true}
              source={
                Object {
                  "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                }
              }
            />,
            <ForwardRef
              color="#f00"
              onPress={[Function]}
              primary={true}
              source={
                Object {
                  "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                }
              }
            />,
          ],
          "color": "#f00",
          "onPress": [Function],
          "primary": true,
          "source": Object {
            "testUri": "../../../src/screens/imagePage/img_avatar3.png",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "center": undefined,
              "children": undefined,
              "color": "#f00",
              "onPress": [Function],
              "primary": true,
              "source": Object {
                "testUri": "../../../src/screens/imagePage/img_avatar3.png",
              },
              "titleColor": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": false,
                "rules": Array [
                  "
    text-align: ",
                  [Function],
                  ";
    color: ",
                  [Function],
                  ";
    font-size:",
                  [Function],
                  ";
",
                ],
              },
              "displayName": "Styled(Component)",
              "propTypes": Object {
                "accessible": [Function],
                "adjustsFontSizeToFit": [Function],
                "allowFontScaling": [Function],
                "disabled": [Function],
                "ellipsizeMode": [Function],
                "minimumFontScale": [Function],
                "nativeID": [Function],
                "numberOfLines": [Function],
                "onLayout": [Function],
                "onLongPress": [Function],
                "onPress": [Function],
                "pressRetentionOffset": [Function],
                "selectable": [Function],
                "selectionColor": [Function],
                "style": [Function],
                "suppressHighlighting": [Function],
                "testID": [Function],
                "textBreakStrategy": [Function],
              },
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "center": undefined,
              "children": undefined,
              "color": "#f00",
              "onPress": [Function],
              "primary": true,
              "source": Object {
                "testUri": "../../../src/screens/imagePage/img_avatar3.png",
              },
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": false,
                "rules": Array [
                  "
    text-align: ",
                  [Function],
                  ";
    color: ",
                  [Function],
                  ";
    padding-top:10;
",
                ],
              },
              "displayName": "Styled(Component)",
              "propTypes": Object {
                "accessible": [Function],
                "adjustsFontSizeToFit": [Function],
                "allowFontScaling": [Function],
                "disabled": [Function],
                "ellipsizeMode": [Function],
                "minimumFontScale": [Function],
                "nativeID": [Function],
                "numberOfLines": [Function],
                "onLayout": [Function],
                "onLongPress": [Function],
                "onPress": [Function],
                "pressRetentionOffset": [Function],
                "selectable": [Function],
                "selectionColor": [Function],
                "style": [Function],
                "suppressHighlighting": [Function],
                "testID": [Function],
                "textBreakStrategy": [Function],
              },
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              "
    width: ",
              [Function],
              ";
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
    padding: 12px;
",
            ],
          },
          "displayName": "Styled(Component)",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
    border: 1px solid  ",
          [Function],
          ";
    width: ",
          [Function],
          ";
    height: ",
          [Function],
          ";
    border-radius: ",
          [Function],
          ";
    flex: 1;
    box-shadow: 10px 5px 5px #000;
",
        ],
      },
      "displayName": "Styled(TouchableOpacity)",
      "getDefaultProps": [Function],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeOpacity": 0.8,
        "children": Array [
          <ForwardRef
            source={
              Object {
                "testUri": "../../../src/screens/imagePage/img_avatar3.png",
              }
            }
          />,
          <ForwardRef
            color="#f00"
            onPress={[Function]}
            primary={true}
            source={
              Object {
                "testUri": "../../../src/screens/imagePage/img_avatar3.png",
              }
            }
          >
            <ForwardRef
              color="#f00"
              onPress={[Function]}
              primary={true}
              source={
                Object {
                  "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                }
              }
            />
            <ForwardRef
              color="#f00"
              onPress={[Function]}
              primary={true}
              source={
                Object {
                  "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                }
              }
            />
          </ForwardRef>,
        ],
        "color": "#f00",
        "onPress": [Function],
        "primary": true,
        "source": Object {
          "testUri": "../../../src/screens/imagePage/img_avatar3.png",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "imageHeight": undefined,
            "imageWidth": undefined,
            "source": Object {
              "testUri": "../../../src/screens/imagePage/img_avatar3.png",
            },
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "rules": Array [
                "
    border-radius: ",
                [Function],
                ";
    width: ",
                [Function],
                ";
    height: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "Styled(Component)",
            "getSize": [Function],
            "prefetch": [Function],
            "propTypes": Object {
              "accessibilityLabel": [Function],
              "accessible": [Function],
              "blurRadius": [Function],
              "capInsets": [Function],
              "defaultSource": [Function],
              "onError": [Function],
              "onLayout": [Function],
              "onLoad": [Function],
              "onLoadEnd": [Function],
              "onLoadStart": [Function],
              "onPartialLoad": [Function],
              "onProgress": [Function],
              "resizeMethod": [Function],
              "resizeMode": [Function],
              "source": [Function],
              "style": [Function],
              "testID": [Function],
            },
            "render": [Function],
            "resolveAssetSource": [Function],
            "styledComponentId": "sc-htpNat",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                color="#f00"
                onPress={[Function]}
                primary={true}
                source={
                  Object {
                    "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                  }
                }
              />,
              <ForwardRef
                color="#f00"
                onPress={[Function]}
                primary={true}
                source={
                  Object {
                    "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                  }
                }
              />,
            ],
            "color": "#f00",
            "onPress": [Function],
            "primary": true,
            "source": Object {
              "testUri": "../../../src/screens/imagePage/img_avatar3.png",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "center": undefined,
                "children": undefined,
                "color": "#f00",
                "onPress": [Function],
                "primary": true,
                "source": Object {
                  "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                },
                "titleColor": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "rules": Array [
                    "
    text-align: ",
                    [Function],
                    ";
    color: ",
                    [Function],
                    ";
    font-size:",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "Styled(Component)",
                "propTypes": Object {
                  "accessible": [Function],
                  "adjustsFontSizeToFit": [Function],
                  "allowFontScaling": [Function],
                  "disabled": [Function],
                  "ellipsizeMode": [Function],
                  "minimumFontScale": [Function],
                  "nativeID": [Function],
                  "numberOfLines": [Function],
                  "onLayout": [Function],
                  "onLongPress": [Function],
                  "onPress": [Function],
                  "pressRetentionOffset": [Function],
                  "selectable": [Function],
                  "selectionColor": [Function],
                  "style": [Function],
                  "suppressHighlighting": [Function],
                  "testID": [Function],
                  "textBreakStrategy": [Function],
                },
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "center": undefined,
                "children": undefined,
                "color": "#f00",
                "onPress": [Function],
                "primary": true,
                "source": Object {
                  "testUri": "../../../src/screens/imagePage/img_avatar3.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-EHOje",
                  "isStatic": false,
                  "rules": Array [
                    "
    text-align: ",
                    [Function],
                    ";
    color: ",
                    [Function],
                    ";
    padding-top:10;
",
                  ],
                },
                "displayName": "Styled(Component)",
                "propTypes": Object {
                  "accessible": [Function],
                  "adjustsFontSizeToFit": [Function],
                  "allowFontScaling": [Function],
                  "disabled": [Function],
                  "ellipsizeMode": [Function],
                  "minimumFontScale": [Function],
                  "nativeID": [Function],
                  "numberOfLines": [Function],
                  "onLayout": [Function],
                  "onLongPress": [Function],
                  "onPress": [Function],
                  "pressRetentionOffset": [Function],
                  "selectable": [Function],
                  "selectionColor": [Function],
                  "style": [Function],
                  "suppressHighlighting": [Function],
                  "testID": [Function],
                  "textBreakStrategy": [Function],
                },
                "render": [Function],
                "styledComponentId": "sc-EHOje",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                "
    width: ",
                [Function],
                ";
    background-color: ",
                [Function],
                ";
    color: ",
                [Function],
                ";
    padding: 12px;
",
              ],
            },
            "displayName": "Styled(Component)",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
    border: 1px solid  ",
            [Function],
            ";
    width: ",
            [Function],
            ";
    height: ",
            [Function],
            ";
    border-radius: ",
            [Function],
            ";
    flex: 1;
    box-shadow: 10px 5px 5px #000;
",
          ],
        },
        "displayName": "Styled(TouchableOpacity)",
        "getDefaultProps": [Function],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
